### PR TITLE
Miscellaneous doc fixes and improvements

### LINKED ISSUE #1142: `PauliExpPairBox` missing from API docs

### PR DESCRIPTION
**Pull request by cqc-alec**
Re. #1155 #1160 .
Fixes #1142 .

### CODE REVIEW

File: pytket/binders/passes.cpp [hunk 1]
```diff
@@ -433,6 +435,8 @@ PYBIND11_MODULE(passes, m) {
       "ThreeQubitSquash", &ThreeQubitSquash,
       "Squash three-qubit subcircuits into subcircuits having fewer CX gates, "
       "when possible, and apply Clifford simplification."
+      "\n\nThe circuit to which this is applied must consist of single-qubit, "
```
[the remark this answers is not in the document]
Comment (CalMacCQ) [reply pytket/binders/passes.cpp:438]: Maybe single qubit gates, CX gate and classical `OpType`s/operations is a bit more explicit.

I've heard the term classical gates used for different things in the past. Like the {X, CnX} gateset.

[the remark this answers is not in the document]
Comment (cqc-alec) [reply pytket/binders/passes.cpp:438]: It means the pure-classical ops, i.e. ClassicalTransform, SetBits etc. I'll change the wording.

[the remark this answers is not in the document]
Comment (CalMacCQ) [reply pytket/binders/passes.cpp:438]: Strange... The additonal line you've added doesn't appear if I build the docs locally on your branch.


Screenshot 2023-12-05 at 14 20 54

I've found the pytket docs build a bit flaky in the past so could be an issue in my local enviornment.

[the remark this answers is not in the document]
Comment (CalMacCQ) [reply pytket/binders/passes.cpp:438]: Ah. Ignore the above. Didn't install a develop version of pytket before building.

**Review by CalMacCQ — APPROVED** [review]
Looks good to me. Thanks Alec :)

**Pull request merged**